pkg/loader/compose/compose.go

```diff
@@ -312,6 +480,125 @@ func (c *Compose) LoadFile(files []string) (kobject.KomposeObject, error) {
+		serviceConfig.ContainerName = composeServiceConfig.ContainerName
+		serviceConfig.Command = composeServiceConfig.Entrypoint
+		serviceConfig.Args = composeServiceConfig.Command
+
+		// This is a bit messy since we use yaml.MemStringorInt
+		// TODO: Refactor yaml.MemStringorInt in kobject.go to int64
+		// Since Deploy.Resources.Limits does not initialize, we must check type Resources before continuing
+		if (composeServiceConfig.Deploy.Resources != types.Resources{}) {
+			serviceConfig.MemLimit = libcomposeyaml.MemStringorInt(composeServiceConfig.Deploy.Resources.Limits.MemoryBytes)
+		}
+
+		// POOF. volumes_From is gone in v3. docker/cli will error out of volumes_from is added in v3
+		// serviceConfig.VolumesFrom = composeServiceConfig.VolumesFrom
+
+		// TODO
+		// Build is not yet supported, see:
+		// https://github.com/docker/cli/blob/master/cli/compose/types/types.go#L9
+		// We will have to *manually* add this / parse.
+		// serviceConfig.Build = composeServiceConfig.Build.Context
+		// serviceConfig.Dockerfile = composeServiceConfig.Build.Dockerfile
+
+		// Gather the environment values
+		// DockerCompose uses map[string]*string while we use []string
+		// So let's convert that using this hack
+		var envKeys []string
+		for key := range composeServiceConfig.Environment {
+			envKeys = append(envKeys, key)
+		}
+		envs := loadEnvVars(envKeys)
+		serviceConfig.Environment = envs
+
+		// Parse the ports
+		// v3 uses a new format called "long syntax" starting in 3.2
+		// https://docs.docker.com/compose/compose-file/#ports
+		serviceConfig.Port = loadV3Ports(composeServiceConfig.Ports)
+
+		// Parse the volumes
+		// Again, in v3, we use the "long syntax" for volumes in terms of parsing
+		// https://docs.docker.com/compose/compose-file/#long-syntax-2
+		serviceConfig.Volumes = loadV3Volumes(composeServiceConfig.Volumes)
+
+		// Label handler
+		// Labels used to influence conversion of kompose will be handled
+		// from here for docker-compose. Each loader will have such handler.
+		for key, value := range composeServiceConfig.Labels {
+			switch key {
+			case "kompose.service.type":
+				serviceType, err := handleServiceType(value)
+				if err != nil {
+					return kobject.KomposeObject{}, errors.Wrap(err, "handleServiceType failed")
+				}
+
+				serviceConfig.ServiceType = serviceType
+			case "kompose.service.expose":
+				serviceConfig.ExposeService = strings.ToLower(value)
+			}
+		}
+
+		// Log if the name will been changed
+		if normalizeServiceNames(name) != name {
+			log.Infof("Service name in docker-compose has been changed from %q to %q", name, normalizeServiceNames(name))
+		}
+
+		// Final step, add to the array!
+		komposeObject.ServiceConfigs[normalizeServiceNames(name)] = serviceConfig
+	}
+
+	return komposeObject, nil
+}
+
+// Uses libcompose's APIProject type and converts it to a Kompose object for us to understand
+func libComposeToKomposeMapping(composeObject *project.Project) (kobject.KomposeObject, error) {
```

Comment:
regardless of what we are using to parse the docker-compose file, we are still reading docker-compose file, so this name `libComposeToKomposeMapping` vs `dockerComposeToKomposeMapping` is really confusing, since they are doing the same thing but again the functions with similar functionality are named by it's implementation over what it does!